api/src/main/java/org/openmrs/Drug.java

```diff
@@ -21,36 +21,69 @@
 import org.hibernate.search.annotations.IndexedEmbedded;
 import org.openmrs.api.context.Context;
 
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
 /**
  * Drug
  */
 @Indexed
+@Entity
+@Table(name = "drug")
 public class Drug extends BaseChangeableOpenmrsMetadata {
 	
 	public static final long serialVersionUID = 285L;
-	
+
 	// Fields
+	@Id
 	@DocumentId
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
```

Comment:
@alinmihaila99 OpenMRS relies on liquibase to create the sequences and then maps these sequences to tables with hibernate. For MySQL its just autoincrement but for db relying on tables for sequences like postgresql it creates a [separate table for sequence](https://stackoverflow.com/questions/44744365/liquibase-postgresql-spring-jpa-id-auto-increment-issue). These sequence tables need to be properly mapped to id columns via annotations. For drug the name of sequence is [drug_drug_id_seq](https://github.com/openmrs/openmrs-core/blob/ef94f268f40e397c2e61dcd1d4b7b6147fa2adbf/api/src/main/resources/org/openmrs/api/db/hibernate/Drug.hbm.xml#L21) . Since we are now extending support to other db's as well, I think we should make this generic and properly specify and map the sequence with the created sequence else we won't be able to extend support to other db's . Also, I am working on extending support to PostgreSQL so this will become a blocker for me. 

This might help you : https://thorben-janssen.com/hibernate-tips-use-custom-sequence/ 
cc. @dkayiwa